app.ts

```diff
@@ -0,0 +1,366 @@
+
+	}
+
+	// Set trim to next data
+	if (btnName == "next" && state.trimEnd != RECORDCOUNT - 1) {
+		// If next page is end of data && there are not enough records to fill window
+		if ((RECORDCOUNT - 1) - (state.trimEnd + 1) < state.records) {
+			state.trimEnd = RECORDCOUNT - 1;
+			state.trimStart = state.trimEnd - state.records + 1;
+		}
+		else {
+			state.trimStart = state.trimEnd + 1;
+			state.trimEnd = state.trimStart + state.records - 1;
+		}
+		isValid = true;
+	}
+
+	// Set trim to end of data
+	if (btnName == "last" && state.trimEnd != RECORDCOUNT - 1) {
+		isValid = true;
+		state.trimEnd = RECORDCOUNT - 1;
+		state.trimStart = state.trimEnd - state.records + 1;
+	}
+
+	if (isValid) {
+		loadIntoTable(el("table"));
+	}
+	else {
+		console.log("Grey out");
+	}
+
+}
+
+
+// Code is only triggered once per user input
+const debounce = (fn: Function, ms: number) => {
+	let timeoutId: ReturnType<typeof setTimeout>;
+	return function (this: any, ...args: any[]) {
+		clearTimeout(timeoutId);
+		timeoutId = setTimeout(() => fn.apply(this, args), ms);
+	};
+};
+
+
+// Last resort debounce application for buttons
+function inputFirst() {
+	pageFunction('first');
+}
+function inputPrev() {
+	pageFunction('prev');
+}
+function inputNext() {
+	pageFunction('next');
+}
+function inputLast() {
+	pageFunction('last');
+}
+const clickFirst = debounce(() => inputFirst(), 800);
+const clickPrev = debounce(() => inputPrev(), 800);
+const clickNext = debounce(() => inputNext(), 800);
+const clickLast = debounce(() => inputLast(), 800);
+
+
+// Add/remove rows from table based on resize event of the window
+window.addEventListener("resize", debounce(async () => {
+
+	// Calculate number rows to be added/deleted
+	let newHeight = Math.floor((window.innerHeight - 160) / 40);
+	let diff = newHeight - state.records;
+
+	let start = state.trimEnd + 1;
+	let end = state.trimEnd + diff;
+
+
+	if (diff < 0) {
+		// Delete rows from last page
+		deleteRows(state.countRec, diff);
+
+		state.countRec = state.countRec + diff;
+		state.trimEnd = state.trimEnd + diff;
+		state.records = newHeight;
+	}
+	else if (diff > 0 && state.trimEnd == RECORDCOUNT - 1) {
+		// Prepend rows as last page gets bigger
+		let timeoutId: ReturnType<typeof setTimeout>;
+
+		end = state.trimStart - 1;
+		start = state.trimStart - diff;
+
+		for (let i = end; i >= start; i--) {
+			let recordsLink = "/records?from=" + i + "&to=" + i;
+			await prependRows(recordsLink);
+		}
+
+
+		state.trimStart = state.trimStart - diff;
+		state.records = newHeight;
+
+	}
+	else if (diff > 0 && end >= RECORDCOUNT) {
+		let addEnd = (RECORDCOUNT - 1) - state.trimEnd;
+		end = RECORDCOUNT - 1;
+		start = end - addEnd + 1;
+		let recordsLink = "/records?from=" + start + "&to=" + end;
+		addRows(recordsLink);
+
+		let addTop = diff - addEnd;
+		end = state.trimStart - 1;
+		start = state.trimStart - addTop;
+
+		for (let i = end; i >= start; i--) {
+			let recordsLink = "/records?from=" + i + "&to=" + i;
```

Comment:
Try to remove the repetition for creating the link string and parsing it as a parameter. 

Rather change the *"*addRows/prependRows**" function to accept the **start** and **end** variables as parameters as well and create the link string inside that function.